invenio_records_resources/services/files/tasks.py

```diff
@@ -39,3 +45,57 @@ def fetch_file(service_id, record_id, file_key):
 
     except FileKeyNotFoundError as e:
         current_app.logger.error(e)
+
+
+# TODO update siegfried signatures (`sf -update`) regularly
+@shared_task(ignore_result=True)
+def detect_file_type(bucket_id, file_key):
+    """Detect the format of the file using siegfried."""
+    # TODO maybe we should go through the Records-Resources files API instead?
+    ov = ObjectVersion.get(bucket_id, file_key)
+    if ov.file is None:
+        return
+
+    # TODO the original filename is lost (renamed to 'data'), but sf uses the filename
+    #      for parts of its algorithm; possible solutions:
+    #      * create a temporary alias (link?) to the file and pass that to sf
+    #      * pipe the file's contents into sf via stdin and use the `-name` arg
+
+    # TODO question: could we utilize siegfried's server mode?
+
+    mimetype, pronom_id = None, None
+    try:
+        sf_bin = "sf"
```

Comment:
if installed with `go install`, the `$GOPATH` needs to be added to the `$PATH` with this logic